cranelift/codegen/src/isa/arm64/inst/emit.rs

```diff
@@ -864,12 +866,20 @@ impl<O: MachSectionOutput> MachInstEmit<O> for Inst {
             &Inst::Nop4 => {
                 sink.put4(0xd503201f);
             }
-            &Inst::Brk { trap_info } => {
-                if let Some((srcloc, code)) = trap_info {
-                    sink.add_trap(srcloc, code);
-                }
+            &Inst::Brk => {
                 sink.put4(0xd4200000);
             }
+            &Inst::Udf { trap_info } => {
+                let (srcloc, code) = trap_info;
+                sink.add_trap(srcloc, code);
+                // TODO remove once Simulator testing is done!
+                if env::var("JSSIM").is_ok() {
```

Comment:
I'd prefer to leave this out if possible, to keep the diff clean. (Question: is there any reason we can't always use the "custom payload"? I'm assuming this is a magic value the simulator recognizes; does the real-hardware case care that the value is zero?)

Reply:
Yes, I'd actually want the arm64 simulator to do the same as the hardware. Note in this case that udf emits a hlt in presence of the arm64 simulator, while it's really an undefined instruction on real hardware. I'll remove this here.